ecosystem/sep-0010.md

```diff
@@ -59,9 +59,15 @@ The authentication flow is as follows:
    the transaction isn't malicious.
 1. The **Client** verifies that the transaction is signed by the **Server Account** obtained through discovery flow.
 1. The **Client** verifies that the transaction's first operation is a Manage Data operation that has its:
-   1. Source account set to the **Client Account**
+   1. Source account set to the **Client Account** or the G-zero address
    1. Key set to `<home domain> auth` where the home domain is the **Home Domain**.
    1. Value set to a nonce value.
+1. If the source account is set to the G-zero address, the **Client** verifies that the transaction has an Invoke Host Function operation with:
+   1. Host Function set to `HOST_FUNCTION_TYPE_INVOKE_CONTRACT` with:
```

Comment:
Operation and transaction are irrelevant for the custom account signer. I think it would make more sense to specify what is the invocation that server has to build and use the auth entry with (i.e. only a signed SorobanAuthEntry is actually necessary to be provided)

Reply:
Given this more thought and I agree. The reason I was avoiding changing the entity to an auth entry only is because there's a bunch of data that the transaction and operatons hold, and we need to retain all that information in the contract auth version.

But most of the information can be encoded in the invocation that's being defined. The invocation arguments can be a map of key-values with each of the things that the transaction currently contains, like the `web_auth_domain` which is important for preventing a forwarding vulnerability.

So the flow would be something like this:
```mermaid
sequenceDiagram
Client ->> Server: Get Challenge
Server ->> Server: Build SorobanAuthorizedInvocation<br>(with web_auth_domain,<br> home_domain,<br> random value,<br> authenticating account,<br> etc...)
Server ->> Server: Sign SorobanAuthorizedInvocation
Server ->> Client: Return SorobanAuthorizedInvocation + (Server Sig)
Client ->> Client: Verify SorobanAuthorizedInvocation
Client ->> Client: Sign SorobanAuthorizedInvocation
Client ->> Client: Build SorobanAddressCredentials
Client ->> Server: Get Token<br/>(with SorobanAuthorizedInvocation,<br>Server Sig,<br>SorobanAddressCredentials)
Server ->> Server: Verify Server Sig
Server ->> Server: Verify Client Sig<br>with Simulation
Server ->> Client: Return Token
```

It's convenient today that everything lives inside a Transaction. But I guess that isn't strictly required. And an advantage of not putting it in a transaction is if we do we have to duplicate all the information into the SorobanAuthorizationInvocation to make sure that the client is signing over it all.